File: CS/CS/Controllers/HomeController.cs
using System;
using System.Web.Mvc;
using CS.Models;

namespace CS.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        /*DataTable*/
        public ActionResult DataTableDataBinding() {
            if(Session["DataTableModel"] == null)
                Session["DataTableModel"] = InMemoryModel.GetDataTableModel();

            return View(Session["DataTableModel"]);
        }
        public ActionResult DataTableDataBindingPartial() {
            return PartialView(Session["DataTableModel"]);
        }
        /**/

        /*Typed List*/
        public ActionResult TypedListDataBinding() {
            if(Session["TypedListModel"] == null)
                Session["TypedListModel"] = InMemoryModel.GetTypedListModel();

            return View(Session["TypedListModel"]);
        }
        public ActionResult TypedListDataBindingPartial() {
            return PartialView(Session["TypedListModel"]);
        }
        /**/
    }
}

File: VB/CS/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports CS.Models

Namespace CS.Controllers

    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            Return View()
        End Function

        ' DataTable
        Public Function DataTableDataBinding() As ActionResult
            If Session("DataTableModel") Is Nothing Then Session("DataTableModel") = InMemoryModel.GetDataTableModel()
            Return View(Session("DataTableModel"))
        End Function

        Public Function DataTableDataBindingPartial() As ActionResult
            Return PartialView(Session("DataTableModel"))
        End Function

        ' 
        ' Typed List
        Public Function TypedListDataBinding() As ActionResult
            If Session("TypedListModel") Is Nothing Then Session("TypedListModel") = InMemoryModel.GetTypedListModel()
            Return View(Session("TypedListModel"))
        End Function

        Public Function TypedListDataBindingPartial() As ActionResult
            Return PartialView(Session("TypedListModel"))
        End Function
    ' 
    End Class
End Namespace
